### PR TITLE
Enable PowerPC

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,9 +8,8 @@ package:
...
 
 source:
-  # the tool is GPL licensed but the download is not possible without registration, so we use a copy here
-  url: https://bitbucket.org/simonbray/ambertools/raw/9e3a9289ca97a2ba0f58cebdfb102c88e8912d9f/AmberTools19.tar.bz2
-  sha256: 8315929b8e1d597907405ddab31d41c556a777f7df7df98745ee9d08c980bae5
+  url: https://ambermd.org/downloads/AmberTools19.tar.bz2
+  sha256: 0c86937904854b64e4831e047851f504ec45b42e593db4ded92c1bee5973e699
 
 build:
   number: 0
```
Comment (simonbray) [inline recipe/meta.yaml]: @jaimergp, should we bump this?

Comment (jaimergp) [reply recipe/meta.yaml]: Now that we might need to change the requirements, yes! Bumped up!

**Comment by simonbray** [comment]
`Error: /bin/csh not found on your system! Install csh or tcsh and rerun configure.`
I fixed with with the patch file included in the OSX PR.

**Comment by jaimergp** [comment]
I managed to reproduce the NetCDF error locally. Attaching log here for reference.

[netcdf_config.log](https://github.com/conda-forge/ambertools-feedstock/files/3984112/netcdf_config.log)

**Comment by jaimergp** [comment]
Further local attempts have been frustrated by a problem during NetCDF compilation :/

[config.log](https://github.com/conda-forge/ambertools-feedstock/files/3984973/config.log)

**Comment by jaimergp** [comment]
The `netcdf-fortran` folder contained a separate log (attached). All the relevant errors point to the same problem:

```
configure:5326: /home/conda/build_artifacts/ambertools_1576779864067/_build_env/bin/powerpc64le-conda_cos7-linux-gnu-gfortran -c -fPIC -fopenmp -mcpu=power8 -mtune=power8 -mpower8-fusion -mpower8-vector -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem /home/conda/build_artifacts/ambertools_1576779864067/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/include -fdebug-prefix-map=/home/conda/build_artifacts/ambertools_1576779864067/work=/usr/local/src/conda/ambertools-19.10 -fdebug-prefix-map=/home/conda/build_artifacts/ambertools_1576779864067/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place=/usr/local/src/conda-prefix -O3 -mtune=native -msse -mfpmath=sse -x f95 conftest.f90 >&5
f951: Error: unrecognized command line option '-msse'
f951: Error: unrecognized command line option '-mfpmath=sse'
configure:5326: $? = 1
configure: failed program was:
|       program main
|
|       end
configure:5335: result: unknown
configure:5338: error: Fortran could not compile .f90 files
```

It tries to use SSE in PowerPC, which is not available. It also overrides some flags already defined by the `conda` compilers (like `mtune`).

**Comment by simonbray** [comment]
Maybe you could avoid compiling netcdf by including it as a dependency and using the `--with-netcdf` flag in the configure script?

**Comment by jaimergp** [comment]
No `ppc64le` available for `libnetcdf` :( I'll have to submit a PR for that one too. Yay!

**Comment by jaimergp** [comment]
Oh wait, `defaults` has it. Let me try! Thanks!

**Comment by jaimergp** [comment]
Updates on my progress:

* I had to patch `AmberTools/src/configure2` to pass `ppc64le` manually to `fftw-3.3` configuration. Some `sse` flag is still around and I couldn't find it, so for now, `-nosse` is being used - it shouldn't have an effect on `ppc64le` anyway
* `boost` fails to build, resulting in two (unimportant?) programs not getting compiled

Ideally, we should patch AmberTools to use the external `boost` and `fftw` libraries (and maybe others?), but I am not familiar with their source and I don't even know where to begin.

**Comment by jaimergp** [comment]
@conda-forge-admin, please restart ci

**Comment by simonbray** [comment]
@conda-forge-admin, please restart ci

**Comment by jaimergp** [comment]
I think that Travis error is unrelated to us. It might have to do with the domain? (.org vs .com)?

**Comment by jaimergp** [comment]
I still need to figure out what's wrong with `boost` now, but that will need some local tests.

**Comment by jaimergp** [comment]
Oh, that did the trick! While getting `ppc64le` to build I made some other changes:

- I have tried not to use the vendored libraries (`lapack`, `blas`, `boost`, `fftw`, `netcdf`) and use conda packages instead for better compatibility. That required patching `config.h` manually after `configure` has run; I hope that's ok and hopefully sets a starting point to get AmberTools to play better with `conda-forge`.
- It also includes the corresponding `activate` / `deactivate` scripts to set `AMBERHOME` automatically.
- `xleap` is being built now.

Some more stuff I would try would be building parallelized versions, maybe with multiple outputs? CUDA and/or MKL support could be added too at some point, if needed.

**Comment by simonbray** [comment]
@jaimergp, shall I merge, or do you have anything else to add?

Also, would you like to add yourself as a maintainer? I'm neither a conda expert nor a AmberTools developer, so I could definitely use some help.

**Comment by jaimergp** [comment]
I would like to test the package locally first. I have changed a few things so we might find errors not caught by the tests. Ideally, we would merge to a branch and upload to a non-main label, but maybe this package is not as popular to need such measures? Shall we ask CF core for feedback? 

And yes, thanks! I will add myself as a maintainer later this weekend.

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Comment by jaimergp** [comment]
There a couple of binaries and libraries that retain the original build path (with all the `_placeholder_placeholder...` stuff) and they are not getting patched by `patchelf`. I have tried with `--prefix` but it didn't work. My next attempt will be using the CMake alternative, which looks promising.

**Comment by simonbray** [comment]
Nice work getting the cmake build to work!

Is it worth trying this for MacOS as well?

**Comment by jaimergp** [comment]
Wow! I didn't expect a successful build so soon. I guess it wouldn't hurt trying with MacOS too. Builds need to be restarted for `ppc64le` anyway. I'll add this in the next push.

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Comment by jaimergp** [comment]
Let's see if that was a temporary error... I can't rerender locally due to some problem with my installation (`libarchive` cannot be found but it's installed).

@conda-forge-admin, please rerender

**Comment by simonbray** [comment]
Try merging this: https://github.com/jaimergp/ambertools-feedstock/pull/1

**Comment by jaimergp** [comment]
Thanks!

**Comment by jaimergp** [comment]
The MacOS error seems to be linked to `ld` (heh, see what I did there?), as commented in https://github.com/conda-forge/superlu_dist-feedstock/pull/4#issuecomment-568512259 (see [error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=104364&view=logs&j=ce7cbaa7-582c-5570-40f4-267b3d85fc9f&t=4a56039b-7269-50ae-de7a-538e91e6de06&l=275)). I'll follow their suggestion and pin `ld` in MacOS for now.

**Comment by jaimergp** [comment]
I think some configuration must be being ignored by the CMake system. I managed to get my OSX VM running again so I'll debug locally now. Thanks for the patience!

**Comment by simonbray** [comment]
The OSX stuff can potentially be done in another PR if you prefer? You already did a lot more here than just PowerPC builds.

**Comment by jaimergp** [comment]
I'd be fine with that. Shall we merge into a different branch that uploads to a different label for now, or just straight to master?

**Comment by simonbray** [comment]
I think straight to master is fine. What alternative label would you propose, if we did it that way?

**Comment by jaimergp** [comment]
`testing_ambertools`, probably. We used `testing` in `openmm`, but turns out other people had the same idea (ie, `matplotlib`), so if you use `-c conda-forge/label/testing` you will pull beta matplotlib packages too (accidentally, I presume). I wouldn't like to release a broken packages (imports and `-h` messages work ok, but we have no clue about the tests).

**Comment by jaimergp** [comment]
We can also wait and build locally for Linux to test the package, with no branches or labels involved. This might be even easier.

**Comment by simonbray** [comment]
So far I just did a few tests locally. But I can see the advantage of testing a package that was built using the conda-forge infrastructure too. Either option is fine with me.

**Comment by jaimergp** [comment]
Did you use the Python script `build-locally.py` in Linux 64? That should be an exact reproduction of the CI environments thanks to the Docker images, so if you tested that, we should be good. If you used a different approach, maybe we should double check with the py script just in case.

**Comment by simonbray** [comment]
Oh, to be clear, I was referring to the original recipe I wrote. I did not test this PR (yet - I can help a bit if you like).

**Comment by jaimergp** [comment]
In the meantime I keep testing OSX. I managed to reproduce the same error locally, and found other issue where they have the same (unsolved?) problem: https://github.com/AnacondaRecipes/aggregate/issues/112

I tried setting the sysroot manually, but it has no effect. We need `cc1` (available through `gcc`, but this is not compiled for `osx`)... might try with the old version?

**Comment by jaimergp** [comment]
I ran some tests locally for Linux 64 and seems to be ok. They are not exhaustive, but main components (cpptraj, antechamber, tleap) look ok! I say we can merge to master after disabling OSX here. I'll re-render in the next commit.

**Comment by simonbray** [comment]
Thanks @jaimergp!

**Comment by jaimergp** [comment]
You are welcome!

**Pull request merged**